log4go_test.go

```diff
@@ -128,9 +105,48 @@ func TestFileLogWriter(t *testing.T) {
 	defer os.Remove(testLogFile)
 
 	w.LogWrite(newLogRecord(CRITICAL, "source", "message"))
+	w.LogWrite(newLogRecord(CRITICAL, "source", "message"))
+	w.Close()
+	runtime.Gosched()
+
+	if contents, err := ioutil.ReadFile(testLogFile); err != nil {
+		t.Errorf("read(%q): %s", testLogFile, err)
+	} else if len(contents) != 100 {
+		t.Errorf("malformed filelog: %q (%d bytes)", string(contents), len(contents))
+	}
+
+	w = NewFileLogWriter(testLogFile, false)
+	w.SetRotate(true)
+	w.SetRotateLines(5)
+
+	// Already has one line from write above
+	if w.maxlines_curlines != 2 {
+		t.Log(w.maxlines_curlines)
+		t.Fatal("maxlines_curlines not initialized properly")
+	}
+
+	if w.maxsize_cursize != 100 {
+		t.Log(w.maxsize_cursize)
+		t.Fatal("maxsize_cursize not initialized properly")
+	}
+
+	for i := 0; i < 4; i++ {
+		w.LogWrite(newLogRecord(CRITICAL, "source", "message"))
+	}
+	defer os.Remove(testLogFile + ".1")
+
+	// Give a bit of time for the writes to finish
+	time.Sleep(time.Second * 1)
```

Comment:
nit: Is there a way to skip the sleep and avoid a future heisenbuild-bug? Could `w.Close()` be updated to `w.file.Sync()`?

Reply:
Sure, let me try it